lib/puppet/type/cisco_route_map.rb

```diff
@@ -0,0 +1,1642 @@
+      set_metric_additive                    => false,
+      set_metric_bandwidth                   => 44,
+      set_metric_delay                       => 55,
+      set_metric_reliability                 => 66,
+      set_metric_effective_bandwidth         => 77,
+      set_metric_mtu                         => 88,
+      set_metric_type                        => 'external',
+      set_nssa_only                          => true,
+      set_origin                             => 'egp',
+      set_path_selection                     => true,
+      set_tag                                => 101,
+      set_vrf                                => 'igp',
+      set_weight                             => 222,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+) (\d+) (\S+)$/,
+      [
+        [:rmname, identity],
+        [:sequence, identity],
+        [:action, identity],
+      ],
+    ]
+    patterns
+  end
+
+  # Overwrites name method. Original method simply returns self[:name],
+  # which is no longer valid or complete.
+  # Would not have failed, but just return nothing useful.
+  def name
+    "#{self[:rmname]} #{self[:sequence]} #{self[:action]}"
+  end
+
+  newparam(:name) do
+    desc 'Name of cisco_route_map, not used, but needed for puppet'
+  end
+
+  newparam(:action, namevar: true) do
+    desc 'Action for set oprtations. Valid values are permit or deny.'
+    munge(&:to_s)
+    newvalues(:permit, :deny)
+  end # param action
+
+  newparam(:sequence, namevar: true) do
+    desc "Sequence to insert/delete existing route-map entry.
+          Valid values are integer."
+  end # param sequence
+
+  newparam(:rmname, namevar: true) do
+    desc 'Name of the route map instance. Valid values are string.'
+  end # param rmname
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:description) do
+    desc "Description of the route-map. Valid values are string,
+          keyword 'default'"
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property description
+
+  newproperty(:match_as_number, array_matching: :all) do
+    format = '[range1, range2]'
+    desc 'Match BGP peer AS number. An array of [range1, range2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
```

Comment:
we need this because of the puppet notice which does not add proper quotes otherwise